src/core/file_format/trp.cpp

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "trp.h"
+#ifdef _WIN64
+#include <Windows.h>
+#elif
+#include <limits.h>
+#include <unistd.h>
+#endif
+
+TRP::TRP() = default;
+TRP::~TRP() = default;
+
+void TRP::GetNPcommID(std::filesystem::path trophyPath, int fileNbr) {
+    std::filesystem::path trpPath = trophyPath / "sce_sys/npbind.dat";
+    Common::FS::IOFile npbindFile(trpPath, Common::FS::FileAccessMode::Read);
+    if (!npbindFile.IsOpen()) {
+        return;
+    }
+    for (int i = 0; i < fileNbr; i++) {
+        std::vector<u8> vec(12);
+        npbindFile.Seek(0x84 + (i * 0x180));
```

Comment:
What is 12 here? Could you make a few constexpr variables here so these are not magic numbers. Also a small comment as to why vector starts out with size 12 and then gets resized to 16

Reply:
npcommid is 12, needed 4 extra zeros for enc/dec